lib/srv/db/proxyserver.go

```diff
@@ -364,9 +397,9 @@ func (s *ProxyServer) Connect(ctx context.Context, proxyCtx *common.ProxyContext
 			ConnType: types.DatabaseTunnel,
 		})
 		if err != nil {
-			// Connection problem indicates reverse tunnel to this server is down.
-			if trace.IsConnectionProblem(err) {
-				s.log.WithError(err).Warnf("Failed to dial %v.", server)
+			// If an agent is down, we'll retry on the next one (if available).
+			if isReverseTunnelDownError(err) {
```

Comment:
Why do we need to check for error type anyway? Shouldn't we just try to connect to all servers in case of any error?

Reply:
Yeah I thought about it originally and decided to only retry reverse tunnel errors specifically to be on the safe side and not retry errors that should not be retried (like, target database connection errors, rbac, etc.). We do it same way for kube and apps.